src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceTestTransformer.java

```diff
@@ -59,20 +76,127 @@ public RubyGapicSurfaceTestTransformer(GapicCodePathMapper rubyPathMapper) {
 
   @Override
   public List<String> getTemplateFileNames() {
-    return ImmutableList.of(SMOKE_TEST_TEMPLATE_FILE);
+    return ImmutableList.of(SMOKE_TEST_TEMPLATE_FILE, UNIT_TEST_TEMPLATE_FILE);
   }
 
   @Override
   public List<ViewModel> transform(Model model, GapicProductConfig productConfig) {
-    List<ViewModel> views = new ArrayList<>();
+    ImmutableList.Builder<ViewModel> views = ImmutableList.builder();
+    views.add(createUnitTestView(model, productConfig));
     for (Interface apiInterface : new InterfaceView().getElementIterable(model)) {
       GapicInterfaceContext context = createContext(apiInterface, productConfig);
       if (context.getInterfaceConfig().getSmokeTestConfig() != null) {
         views.add(createSmokeTestClassView(context));
       }
     }
 
-    return views;
+    return views.build();
+  }
+
+  ///////////////////////////////////// Unit Test ///////////////////////////////////////
+
+  private MockCombinedView createUnitTestView(Model model, GapicProductConfig productConfig) {
+    SurfaceNamer namer = new RubySurfaceNamer(productConfig.getPackageName());
+    ImportSectionView importSection =
+        importSectionTransformer.generateTestImportSection(model, productConfig);
+    return MockCombinedView.newBuilder()
+        .outputPath(UNIT_TEST_OUTPUT_FILE)
+        .serviceImpls(ImmutableList.<MockServiceImplView>of())
+        .mockServices(ImmutableList.<MockServiceUsageView>of())
+        .testClasses(generateTestClasses(model, productConfig, namer))
+        .templateFileName(UNIT_TEST_TEMPLATE_FILE)
+        .fileHeader(fileHeaderTransformer.generateFileHeader(productConfig, importSection, namer))
+        .build();
+  }
+
+  private List<ClientTestClassView> generateTestClasses(
+      Model model, GapicProductConfig productConfig, SurfaceNamer namer) {
+    ImmutableList.Builder<ClientTestClassView> testClasses = ImmutableList.builder();
+    String apiSettingsClassName =
+        namer.getNotImplementedString(
+            "RubyGapicSurfaceTestTransformer.generateTestView - apiSettingsClassName");
```

Comment:
I went with calling it this way since it is the way it is called in many other places. It might be good to do a refactor with some reflection in a separate PR